### DISCUSSION TITLE
save RT for each keystroke when typing using survey-text plugin

### DISCUSSION THREAD

**Discussion by SRJongman** [jspsych/jsPsych#1151] [Q&A]
Hi,

I am trying to program a picture typing experiment where participants see pictures and have to type the label and press Enter to move on the next trial. It works using the survey-text plugin but I need to know the response time of the first key stroke as well as the response time for the Enter button press. Using the standard plugin I only have the latter. Is there a way to adapt the survey-text plugin to have the different letters stored separately in an array, together with the time each letter was pressed or something along those lines? I am mostly interested in the first key stroke but having all information would be even better. Now it just save the whole text in each field after the participant presses the submit button/Enter and I am not sure how to start saving the data before that final trial event. Can the survey-text plugin be adapted to have each key stroke and its RT saved? Is there a different plugin that would be better? 

Kind regards,

Suzanne

**Comment by jodeleeuw** [comment]
Hi @SRJongman ,

This was discussed a bit here: https://github.com/jspsych/jsPsych/discussions/772
